pkg/lib/interaction/nodes/settings_action_end.go

```diff
@@ -0,0 +1,46 @@
+package nodes
+
+import (
+	"github.com/authgear/authgear-server/pkg/lib/interaction"
+	"github.com/authgear/authgear-server/pkg/lib/oauth/oauthsession"
+)
+
+func init() {
+	interaction.RegisterNode(&NodeSettingsActionEnd{})
+}
+
+type EdgeSettingsActionEnd struct{}
+
+func (e *EdgeSettingsActionEnd) Instantiate(ctx *interaction.Context, graph *interaction.Graph, input interface{}) (interaction.Node, error) {
+	return &NodeSettingsActionEnd{}, nil
+}
+
+type NodeSettingsActionEnd struct{}
+
+func (n *NodeSettingsActionEnd) Prepare(ctx *interaction.Context, graph *interaction.Graph) error {
+	return nil
+}
+
+func (n *NodeSettingsActionEnd) GetEffects() ([]interaction.Effect, error) {
+	return []interaction.Effect{
+		interaction.EffectOnCommit(func(ctx *interaction.Context, graph *interaction.Graph, nodeIndex int) error {
+			entry, err := ctx.OAuthSessions.Get(ctx.OAuthSessionID)
+			if err != nil {
+				return err
+			}
+
+			if entry.T.AuthorizationRequest.ResponseType() != "settings_action" {
```

Comment:
No need to check. We can always save the result event if it was not a settings action in the first place.